### ISSUE TITLE
[Android] If a non-must field is left empty in the Issuance Web App, the produced QR-Code is always shown as invalid by the verifier App

### ISSUE THREAD

**Issue by RalicaY** [eu-digital-green-certificates/dgca-verifier-app-android#51]
## Describe the bug

If an optional field is omitted (it is NULL value) the issuer app generates a QR-Code which is always shown as invalid by the verifier app.

For example, a test certificate with negative result and a valid signature, but the name has been left blank.

A sample QR Code and the data to generate it is attached,

## Expected behaviour

A valid certificate should be shown as valid also when optional fields are left empty.

## Technical details

![not_detected](https://user-images.githubusercontent.com/76050122/119681487-68c82c00-be42-11eb-8d3b-1eade6d3f4fe.png)

**Comment by MykhailoNester** [comment]
From what I see in json there are `null` values - `"fn":null`.

```
   "nam":{
      "fnt":"NM",
      "fn":null,
      "gnt":null,
      "gn":null
   }
```

 It is the reason for schema validation fail on client side:
 
  ```
 error: instance type (null) does not match any allowed primitive type (allowed: ["string"])
    level: "error"
    schema: {"loadingURI":"#","pointer":"/$defs/person_name/properties/fn"}
    instance: {"pointer":"/nam/fn"}
    domain: "validation"
    keyword: "type"
    found: "null"
    expected: ["string"]
```

From this schema that we check: 
https://github.com/eu-digital-green-certificates/ehn-dgc-schema/blob/main/DGC.combined-schema.json

we have:

```
"fn":{
   "title":"Family name",
   "description":"The family or primary name(s) of the person addressed in the certificate",
   "type":"string",
   "maxLength":50,
   "examples":[
      "d'Červenková Panklová"
   ]
}
```

`"type":"string",`  - **Error here. This property supports only `String` type**.  

To fix this we need to update schema fields type in `ehn-dgc-schema` repository to this:
`"type": ["null", "string"]` or just omit all null and not include them into json. 

With schema update I see such result for QR code above:

![Screenshot_20210526-205041](https://user-images.githubusercontent.com/9447674/119709011-67f4c180-be65-11eb-87f7-bd207055df0c.png)

**Comment by oleksandrsarapulovgl** [comment]
@SchulzeStTSI could you please clarify what solution will we follow here? Will we update schema, or will you block null value on BE side?

FYI @MykhailoNester @Hendrik-Schmidt-Schierhorn-TSI

**Comment by SchulzeStTSI** [comment]
@oleksandrsarapulovgl In the schema is just the fnt field mandatory. So all others seems to be optional. Can we simply remove null fields before schema checkup?

**Comment by oleksandrsarapulovgl** [comment]
I guess we can, but why would we do that if we can just make it nullable as @MykhailoNester described? @SchulzeStTSI let's make it nullable in the schema?

**Comment by SchulzeStTSI** [comment]
@oleksandrsarapulovgl I will analyze it.

**Comment by SchulzeStTSI** [comment]
I've pushed it to https://github.com/eu-digital-green-certificates/dgca-issuance-web/issues/79 But please correct it as proposed by @MykhailoNester locally in the schema of the verifier. The offical schema can not be modified.

**Comment by oleksandrsarapulovgl** [comment]
PR: https://github.com/eu-digital-green-certificates/dgca-app-core-android/pull/16

**Comment by ltranvan** [comment]
Hello, 
After you have fixed the issue, please shifted it to implemented. A shift to "In Test/Retest" is just only for the case, once a build with a fix is allready created and provided for test/retest. 
As I suppose, this is here not the case, so i will let it in state "Implemented" till a new build for test would be created and provided.

**Comment by oleksandrsarapulovgl** [comment]
Thank you @ltranvan, will do

**Comment by RalicaY** [comment]
Test of the fix was successful in  Version 1.0.2.  The bug is not to be seen.
